src/sdc11073/roles/alarmprovider.py

```diff
@@ -403,35 +381,12 @@ def _getAlertSystemStates_needingUpdate(self):
 
     def _updateAlertSystemState_CurrentAlerts(self):
         """ updates AlertSystemState present alarms list"""
-        alertStatesNeedingUpdate = self._getAlertSystemStates_needingUpdate()
-        if len(alertStatesNeedingUpdate) > 0:
-            try:
-                with self._mdib.mdibUpdateTransaction() as tr:
+        try:
+            with self._mdib.mdibUpdateTransaction() as tr:
+                alertStatesNeedingUpdate = self._getAlertSystemStates_needingUpdate()
+                if len(alertStatesNeedingUpdate) > 0:
                     tr_states = [tr.getAlertState(s.descriptorHandle) for s in alertStatesNeedingUpdate]
                     self._updateAlertSystemStates(self._mdib, tr, tr_states)
-            except:
-                exc = traceback.format_exc()
-                self._logger.error('_checkAlertStates: {}', exc)
-
-    # def _handleDelegateTimeouts(self):
-    #     if self._lastActivateAllDelegableAlerts:
-    #         # find the minimal invocation_effective_timeout
-    #         all_op_descrs = self._mdib.descriptions.NODETYPE.get(namespaces.domTag('SetAlertStateOperationDescriptor'), [])
-    #         timeouts = [op.InvocationEffectiveTimeout for op in all_op_descrs]
-    #         timeouts = [t for t in timeouts if t is not None]
-    #         if not timeouts:
-    #             return # nothing to do
-    #         minimal_invocation_effective_timeout = min(timeouts)
-    #         if time.time() - self._lastActivateAllDelegableAlerts > minimal_invocation_effective_timeout:
-    #             # expired, set all AlertSignalState.ActivationState to 'Off'
-    #             with self._mdib.mdibUpdateTransaction() as mgr:
-    #                 for op in all_op_descrs:
-    #                     signalDescr = self._mdib.descriptions.handle.getOne(op.OperationTarget)
-    #                     allSignalDescriptors = self._mdib.descriptions.ConditionSignaled.get(signalDescr.ConditionSignaled, [])
-    #                     ss = mgr.getAlertState(signalDescr.handle)
-    #                     if ss.ActivationState == AlertActivation.ON:
-    #                         ss.ActivationState = AlertActivation.OFF
-    #                         self._activateFallbackAlertSignals(signalDescr, allSignalDescriptors, mgr)
-    #                     else:
-    #                         mgr.ungetState(ss)
-    #             self._lastActivateAllDelegableAlerts = 0
+        except:
```

Comment:
because the worker thread in alarmprovider shall not terminate if any error occurs. This is a catch-all handling, I think catching everything is correct here.